Failure on call REST API

Hi all, 

I'm making a API call (GET) and already check on API server when I call to localhost:5001/cites, server return cities data, but I don't know it error on client side, here is log from browser :

`failure.js:18 Error: The X-Total-Count header is missing in the HTTP Response. This header is necessary for pagination. If you are using CORS, did you declare X-Total-Count in the Access-Control-Allow-Headers header?

at convertHTTPResponseToREST (http://localhost:3000/static/js/bundle.js:33928:28)
at http://localhost:3000/static/js/bundle.js:33966:21`

and

`failure.js:18 Error: The Content-Range header is missing in the HTTP Response. This header is necessary for pagination. If you are using CORS, did you declare Content-Range in the Access-Control-Allow-Headers header?

at convertHTTPResponseToREST (http://localhost:3000/static/js/bundle.js:33010:28)
at http://localhost:3000/static/js/bundle.js:33037:21`


May someone can help ? 

Thanks

--- a/docs/RestClients.md
+++ b/docs/RestClients.md
@@ -47,6 +47,18 @@ This REST client fits APIs using simple GET parameters for filters and sorting. 
 | `CREATE`       | `POST http://my.api.url/posts/123`
 | `DELETE`       | `DELETE http://my.api.url/posts/123`
 
+**Note**: The simple REST client expects the API to include a `Content-Range` header in the response to `GET_LIST` and `GET_MATCHING` calls. The value must be the total number of resources in the collection. This allows admin-on-rest to know how many pages of resources there are in total, and build the pagination controls.
+
+```
+Content-Range: posts 0-24/319
+```
+
+If your API is on another domain as the JS code, you'll need to whitelist this header with an `Access-Control-Expose-Headers` [CORS](https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS) header.
+
+```
+Access-Control-Expose-Headers: Content-Range
+```
+
 Here is how to use it in your admin:
 
 ```js
@@ -79,6 +91,18 @@ This REST client fits APIs powered by [JSON Server](https://github.com/typicode/
 | `UPDATE`       | `PUT http://my.api.url/posts/123`
 | `CREATE`       | `POST http://my.api.url/posts/123`
 | `DELETE`       | `DELETE http://my.api.url/posts/123`
+
+**Note**: The jsonServer REST client expects the API to include a `X-Total-Count` header in the response to `GET_LIST` and `GET_MATCHING` calls. The value must be the total number of resources in the collection. This allows admin-on-rest to know how many pages of resources there are in total, and build the pagination controls.
+
+```
+X-Total-Count: 319
+```
+
+If your API is on another domain as the JS code, you'll need to whitelist this header with an `Access-Control-Expose-Headers` [CORS](https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS) header.
+
+```
+Access-Control-Expose-Headers: X-Total-Count
+```
 
 Here is how to use it in your admin:
 

--- a/src/rest/jsonServer.js
+++ b/src/rest/jsonServer.js
@@ -89,7 +89,7 @@ export default (apiUrl, httpClient = fetchJson) => {
         switch (type) {
         case GET_LIST:
             if (!headers.has('x-total-count')) {
-                throw new Error('The X-Total-Count header is missing in the HTTP Response. This header is necessary for pagination. If you are using CORS, did you declare X-Total-Count in the Access-Control-Allow-Headers header?');
+                throw new Error('The X-Total-Count header is missing in the HTTP Response. The jsonServer REST client expects responses for lists of resources to contain this header with the total number of results to build the pagination. If you are using CORS, did you declare X-Total-Count in the Access-Control-Expose-Headers header?');
             }
             return {
                 data: json.map(x => x),

--- a/src/rest/simple.js
+++ b/src/rest/simple.js
@@ -102,7 +102,7 @@ export default (apiUrl, httpClient = fetchJson) => {
         switch (type) {
         case GET_LIST:
             if (!headers.has('content-range')) {
-                throw new Error('The Content-Range header is missing in the HTTP Response. This header is necessary for pagination. If you are using CORS, did you declare Content-Range in the Access-Control-Allow-Headers header?');
+                throw new Error('The Content-Range header is missing in the HTTP Response. The simple REST client expects responses for lists of resources to contain this header with the total number of results to build the pagination. If you are using CORS, did you declare Content-Range in the Access-Control-Expose-Headers header?');
             }
             return {
                 data: json.map(x => x),

